### PR TITLE
Remove check for deprectated functions toUpperCase and toLowerCase

### LINKED ISSUE #6343: `ImplicitDefaultLocale` rule should not check for deprecated functions
## Expected Behavior of the rule
The `ImplicitDefaultLocale` rule should not check for usage of `toUpperCase` and `toLowerCase`.

## Context
The extension functions `String.toUpperCase` and `String.toLowerCase` have been deprecated in favor of `uppercase` and `lowercase`. Both function by default use the `Locale.ROOT` and make the check obsolete

### PR DESCRIPTION
**Pull request by Gosunet**
`toUpperCase` and `toLowerCase` are deprecated, the `ImplicitDefaultLocale` should not check usage for them anymore

Closes #6343

### CODE REVIEW

**Review by schalkms — APPROVED** [review]

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocale.kt [hunk 1]
```diff
@@ -33,17 +30,13 @@ import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
  * "Timestamp: %d".format(System.currentTimeMillis())
  *
  * val str: String = getString()
```
Comment (cortinico) [inline detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocale.kt:32]: Please remove this line also

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocale.kt [hunk 2]
```diff
@@ -33,17 +30,13 @@ import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
...
- * str.toUpperCase()
- * str.toLowerCase()
  * </noncompliant>
  *
  * <compliant>
  * String.format(Locale.US, "Timestamp: %d", System.currentTimeMillis())
  * "Timestamp: %d".format(Locale.US, System.currentTimeMillis())
  *
  * val str: String = getString()
```
Comment (cortinico) [inline detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocale.kt:39]: Please remove this line also

**Review by cortinico — APPROVED** [review]
LGTM modulo a minor nit

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocale.kt (hunk 1 above)
Comment (Gosunet) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocale.kt]: Done!

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocale.kt (hunk 2 above)
Comment (Gosunet) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocale.kt]: ✅

**Review by marschwar — APPROVED** [review]

**Pull request merged**